matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/failure/Extensions.kt

```diff
@@ -73,3 +74,11 @@ fun Throwable.toRegistrationFlowResponse(): RegistrationFlowResponse? {
         null
     }
 }
+
+fun Throwable.isRegistrationAvailabilityError(): Boolean {
+    return this is Failure.ServerError
+            && (this.error.code == MatrixError.M_USER_IN_USE
+                    || this.error.code == MatrixError.M_INVALID_USERNAME
+                    || this.error.code == MatrixError.M_EXCLUSIVE
+                    || this.httpCode == 429)
```

Comment:
429 should not be checked here. But you can can add `&& httpCode == 400`. Also `this.` is not necessary.

Reply:
done in https://github.com/vector-im/element-android/pull/3084